frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/create-local-volume-set-step/selected-capacity.tsx

```diff
@@ -59,84 +50,66 @@ const addNodesOnAvailableDisks = (disks: DiskMetadata[], node: string) =>
-        {
-          key: LABEL_SELECTOR,
-          operator: LABEL_OPERATOR,
-          values: allLvsNodes,
-        },
-      ],
-    },
-  };
-
+export const SelectedCapacity: React.FC<SelectedCapacityProps> = ({
+  state,
+  nodes,
+  lvdResults,
+  dispatch,
+}) => {
   const { t } = useTranslation();
-  const [lvdResults, lvdResultsLoaded, lvdResultsLoadError] = useK8sWatchResource<
-    LocalVolumeDiscoveryResultKind[]
-  >(lvdResultResource);
   const [showNodeList, setShowNodeList] = React.useState(false);
   const [showDiskList, setShowDiskList] = React.useState(false);
 
-  let filteredDisks: DiscoveredDisk[] = [];
-
   const minSize: number = state.minDiskSize
     ? Number(convertToBaseValue(`${state.minDiskSize} ${state.diskSizeUnit}`))
     : 0;
   const maxSize: number = state.maxDiskSize
     ? Number(convertToBaseValue(`${state.maxDiskSize} ${state.diskSizeUnit}`))
     : undefined;
 
-  const allDiscoveredDisks: DiscoveredDisk[] = React.useMemo(() => {
-    if (!lvdResultsLoadError && lvdResultsLoaded && allLvsNodes.length === lvdResults.length) {
-      setIsLoadingDonutChart(false);
-      return createDiscoveredDiskData(lvdResults);
-    }
-    return [];
-  }, [allLvsNodes.length, lvdResults, lvdResultsLoadError, lvdResultsLoaded]);
-
-  if (allDiscoveredDisks.length) {
-    filteredDisks = allDiscoveredDisks.filter(
-      (disk: DiscoveredDisk) =>
-        state.isValidDiskSize &&
-        isValidSize(disk, minSize, maxSize) &&
-        isValidDiskProperty(disk, DISK_TYPES[state.diskType]?.property) &&
-        isValidDeviceType(disk, state.deviceType),
-    );
-  }
+  const allDiscoveredDisks: DiscoveredDisk[] = React.useMemo(
+    () => createDiscoveredDiskData(lvdResults),
+    [lvdResults],
+  );
 
-  const chartDisks = state.lvsIsSelectNodes
-    ? filteredDisks.filter((disk: DiscoveredDisk) => selectedLvsNodes.includes(disk.node))
-    : filteredDisks;
-  const chartNodes: Set<string> = chartDisks.reduce(
-    (nodes: Set<string>, disk: DiscoveredDisk) => nodes.add(disk.node),
-    new Set(),
+  const filteredDisks: DiscoveredDisk[] = React.useMemo(
+    () =>
+      allDiscoveredDisks.length
+        ? allDiscoveredDisks.filter(
+            (disk: DiscoveredDisk) =>
+              state.isValidDiskSize &&
+              isValidSize(disk, minSize, maxSize) &&
+              isValidDiskProperty(disk, DISK_TYPES[state.diskType]?.property) &&
+              isValidDeviceType(disk, state.deviceType),
+          )
+        : [],
+    [allDiscoveredDisks, maxSize, minSize, state.deviceType, state.diskType, state.isValidDiskSize],
   );
 
-  if (!_.isEqual(chartNodes, state.chartNodes)) {
-    dispatch({
-      type: 'wizard/setCreateLocalVolumeSet',
-      payload: { field: 'chartNodes', value: chartNodes },
-    });
-  }
+  const chartDisks = React.useMemo(() => {
+    const selectedNodes = nodes.map((node) => node.name);
+    return filteredDisks.filter((disk: DiscoveredDisk) => selectedNodes.includes(disk.node));
```

Comment:
IMO, it will be better to use `set` instead of `list` on Line 97... searching complexity will reduce a lot in Line 98 ??

Reply:
good catch, I missed n^2 on L98. Thanks